File: Suspension matlab/Brake_force/Brake_force.m
%% 煞車力

clc; clear; close all;
scriptDir = fileparts(mfilename('fullpath'));
addpath(scriptDir);
params = load_data(fullfile(scriptDir,'data.csv'));
paramNames = fieldnames(params);
for paramIdx = 1:numel(paramNames)
    eval([paramNames{paramIdx} ' = params.(paramNames{paramIdx});']);
end
g = brake_g;

fprintf("\n F_driver = %.2f\n", F_driver);

F_mc = F_driver*PR;

fprintf("\n F_mc = %.2f\n", F_mc);
% balance_bar
bb_r = 1-balance_bar;
bb_f = balance_bar;

% 主缸面積
A_mc_f = pi * (D_mc_f / 2)^2;
A_mc_r = pi * (D_mc_r / 2)^2;

P_mc_f = F_mc*bb_f/A_mc_f;
P_mc_r = F_mc*bb_r/A_mc_r;


% 卡鉗面積
A_caliper_f = N_caliper_f * pi * (D_caliper_f / 2)^2;
A_caliper_r = N_caliper_r * pi * (D_caliper_r / 2)^2;

% 主缸力
F_caliper_f = P_mc_f * mu_pad * A_caliper_f;
F_caliper_r = P_mc_r * mu_pad * A_caliper_r;

r_disc = r_disc_o / 2 - d_gap;

F_brake_f = F_caliper_f*r_disc*2/r_w;
F_brake_r = F_caliper_r*r_disc*2/r_w;

fprintf("\n F_brake_f = %.2f\n", F_brake_f);
fprintf("\n F_brake_r = %.2f\n", F_brake_r);

F_brake = F_brake_r+F_brake_f;
fprintf("\n F_brake = %.2f\n", F_brake);

File: Suspension matlab/Brake_force/Brake_force.py
from pathlib import Path
import sys

import numpy as np

ROOT = next(p for p in Path(__file__).resolve().parents if (p / "suspension_utils.py").exists())
sys.path.insert(0, str(ROOT))

from suspension_utils import load_data


def main() -> None:
    params = load_data(Path(__file__).with_name("data.csv"))
    globals().update(params)

    print(f"\n F_driver = {F_driver:.2f}")

    f_mc = F_driver * PR
    print(f"\n F_mc = {f_mc:.2f}")

    bb_r = 1 - balance_bar
    bb_f = balance_bar

    a_mc_f = np.pi * (D_mc_f / 2) ** 2
    a_mc_r = np.pi * (D_mc_r / 2) ** 2

    p_mc_f = f_mc * bb_f / a_mc_f
    p_mc_r = f_mc * bb_r / a_mc_r

    a_caliper_f = N_caliper_f * np.pi * (D_caliper_f / 2) ** 2
    a_caliper_r = N_caliper_r * np.pi * (D_caliper_r / 2) ** 2

    f_caliper_f = p_mc_f * mu_pad * a_caliper_f
    f_caliper_r = p_mc_r * mu_pad * a_caliper_r

    r_disc = r_disc_o / 2 - d_gap
    f_brake_f = f_caliper_f * r_disc * 2 / r_w
    f_brake_r = f_caliper_r * r_disc * 2 / r_w

    print(f"\n F_brake_f = {f_brake_f:.2f}")
    print(f"\n F_brake_r = {f_brake_r:.2f}")
    print(f"\n F_brake = {f_brake_f + f_brake_r:.2f}")


if __name__ == "__main__":
    main()
